apps/web/app/environments/[environmentId]/surveys/[surveyId]/responses/SingleResponse.tsx

```diff
@@ -34,6 +40,14 @@ function findEmail(person) {
 export default function SingleResponse({ data, environmentId }: OpenTextSummaryProps) {
   const email = data.person && findEmail(data.person);
   const displayIdentifier = email || data.personId;
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
+
+  const handleDeleteSubmission = async () => {
+    const deleteResponse = await deleteSubmission(environmentId, data?.surveyId, data?.id);
+    if(deleteResponse?.id?.length > 0)
+    toast.success("Submission deleted successfully.");
+    setDeleteDialogOpen(false);
```

Comment:
after this line the submission was deleted in the backend, but is still shown in the frontend. We are using `SWR` in the frontend the handle the state of the responses.
To use this you need to import SWR in the Single response file and get the current responses:

```
const { responsesData, mutateResponses } = useResponses(environmentId, surveyId);
```

and in the handleDeleteSubmission you need to use the `mutateRespones` to change the current responses. The easiest way would be to just call `mutateResponses()` because then SWR makes a check to the backend and gets the current state of responses again.

Reply:
Yes, that was a miss. 

Thanks for pointing out.
I've fixed this.
Thanks